custom_components/tuya_local/devices/thermoval_tvt40_thermostat.yaml

```diff
@@ -0,0 +1,315 @@
+name: Thermoval WiFi Thermostat
+products:
+  - id: jIRwsT5yLPJO41Po
+    name: TVT 40
+primary_entity:
+  entity: climate
+  translation_key: thermostat
+  dps:
+    - id: 2
+      name: temperature
+      type: integer
+      range:
+        min: 50
+        max: 400
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: Smart
+          value: program
+          icon: "mdi:home-thermometer"
+        - dps_val: Anti_frozen
+          value: away
+          icon: "mdi:snowflake-melt"
+        - dps_val: Manual
+          value: manual
+          icon: "mdi:cursor-pointer"
+    - id: 4
+      name: fault
+      type: integer
+    - id: 5
+      name: room_temperature
+      type: integer
+      mapping:
+        - scale: 10
+    - id: 6
+      name: floor_temp_limit
+      type: integer
+      mapping:
+        - scale: 10
+    - id: 7
+      name: installation
+      type: boolean
+      mapping:
+        - dps_val: true
+          value: Office
+        - dps_val: false
+          value: Home
+    - id: 8
+      name: floor_temperature
+      type: integer
+      mapping:
+        - scale: 10
+    - id: 9
+      name: hvac_mode
+      type: boolean
+      readonly: true

```

Comment:
hvac_mode should not be read-only.  If this is really read-only, rename this attribute (I guess the original tuya name is work_mode, which would be appropriate) to avoid buttons that cannot be used in the UI (the mode will then always show as auto).